app/src/org/commcare/views/notifications/NotificationMessageFactory.java

```diff
@@ -211,7 +227,18 @@ public static NotificationMessage message(MessageTag message, String[] parameter
                 //No big deal, key doesn't need to exist
             }
 
-            return new NotificationMessage(customCategory, title, detail, action, new Date());
+            NotificationActionButtonInfo buttonInfo = null;
+            try {
+                String button = parameters[3] == null ? Localization.get(base + ".button") : Localization.get(base + ".button", new String[]{parameters[3]});
+                buttonInfo = new NotificationActionButtonInfo(button, buttonAction);
+            } catch (Exception e) {
+                if(buttonAction != NotificationActionButtonInfo.ButtonAction.NONE) {
```

Comment:
wrap the `String button` evaluation above under this `if` instead of adding in the exception catcher and then remove the `catch` block completely from here ?